org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPDeleteParticipant.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Vegard IT GmbH and others.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Sebastian Thomschke (Vegard IT GmbH) - initial implementation
+ *******************************************************************************/
+package org.eclipse.lsp4e.operations.rename;
+
+import static org.eclipse.lsp4e.internal.NullSafetyHelper.*;
+
+import java.net.URI;
+import java.util.List;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.lsp4e.LSPEclipseUtils;
+import org.eclipse.lsp4e.LanguageServerWrapper;
+import org.eclipse.lsp4j.DeleteFilesParams;
+import org.eclipse.lsp4j.FileDelete;
+import org.eclipse.lsp4j.FileOperationsServerCapabilities;
+import org.eclipse.ltk.core.refactoring.Change;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+import org.eclipse.ltk.core.refactoring.participants.CheckConditionsContext;
+import org.eclipse.ltk.core.refactoring.participants.DeleteParticipant;
+
+public class LSPDeleteParticipant extends DeleteParticipant {
+
+	private URI oldURI = lateNonNull();
+	private List<LanguageServerWrapper> servers = lateNonNull();
+
+	@Override
+	public String getName() {
+		return "LSP4E Delete"; //$NON-NLS-1$
+	}
+
+	@Override
+	protected boolean initialize(final Object element) {
+		if (element instanceof IFile || element instanceof IFolder) {
```

Comment:
```suggestion
		if (element instanceof IResource res && (res instanceof IFile || res instanceof IFolder)) {
```
and then remove the manual cast?

Same for the other participants